client/src/pages/Property.tsx

```diff
@@ -29,17 +33,46 @@ const Property = () => {
     updatedAt: new Date(),
   });
   const [comments, setComments] = useState<Comment[]>([]);
-
+  const navigate = useNavigate();
   useEffect(() => {
-    getPostComments().then((data) => {
-      setComments(data);
-    });
+    setLoadingPost(true);
+    setError(null);
+    getPost()
+      .then((data) => {
+        setPost(data.post);
+        setLoadingPost(false);
+      })
+      .catch(() => {
+        setLoadingPost(false);
+        setError("An error occurred.");
+      });
+    setLoadingComments(true);
+    setError(null);
+    getPostComments()
+      .then((data) => {
+        setComments(data);
+        setLoadingComments(false);
+      })
+      .catch(() => {
+        setLoadingComments(false);
+        setError("An error occurred.");
+      });
```

Comment:
You are calling the API twice to get the same information - `comments` is already a part of the JSON object returned by the API, you can access them via `data.post.comments`

Reply:
The function `getPostComments` isn't needed either, same with `loadingComments` state.